File: common/src/main/java/top/trumeet/common/utils/ElapsedTimer.java
package top.trumeet.common.utils;

import android.os.SystemClock;

public class ElapsedTimer {
    private long start = time();

    public long start() {
        return start = time();
    }

    public long restart() {
        long old = start;
        start = time();
        return start - old;
    }

    public long elapsed() {
        return time() - start;
    }

    private long time() {
        return SystemClock.elapsedRealtime();
    }
}

File: common/src/main/java/top/trumeet/common/utils/ElapsedTimer.kt
package top.trumeet.common.utils

import android.os.SystemClock

class ElapsedTimer {
    private var start = time()

    fun start(): Long {
        start = time()
        return start
    }

    fun restart(): Long {
        val old = start
        start = time()
        return start - old
    }

    fun elapsed(): Long {
        return time() - start
    }

    private fun time(): Long {
        return SystemClock.elapsedRealtime()
    }
}
